htdocs/xoops_lib/Xoops/Core/Security.php

```diff
@@ -189,11 +189,12 @@ public function checkBadips()
             && $_SERVER['REMOTE_ADDR'] != ''
         ) {
             foreach ($xoops->getConfig('bad_ips') as $bi) {
-                if (!empty($bi) && preg_match('/' . $bi . '/', $_SERVER['REMOTE_ADDR'])) {
-                    exit();
+                if (!empty($bi) && false !== strpos($_SERVER['REMOTE_ADDR'],$bi)) {
+                    return true;
```

Comment:
The intent here is to kill the session, white screen with exit(), immediately if a bad IP is recognized. I know that makes it impossible to test, but we have to live with that.